tests/Unit/Domain/FunctionsOfTime/Test_ReadSpecThirdOrderPiecewisePolynomial.cpp

```diff
@@ -8,88 +8,66 @@
 #include "Domain/FunctionsOfTime/Tags.hpp"
 #include "Domain/Structure/ElementId.hpp"
-#include "Framework/ActionTesting.hpp"
+#include "Framework/TestCreation.hpp"
 #include "IO/H5/AccessType.hpp"
 #include "IO/H5/Dat.hpp"
 #include "IO/H5/File.hpp"
-#include "IO/Importers/Actions/ReadSpecThirdOrderPiecewisePolynomial.hpp"
-#include "IO/Importers/Tags.hpp"
 #include "Informer/InfoFromBuild.hpp"
+#include "Options/Auto.hpp"
 #include "Options/Options.hpp"
 #include "Options/ParseOptions.hpp"
-#include "Parallel/PhaseDependentActionList.hpp"
 #include "Utilities/FileSystem.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/TMPL.hpp"
 
 namespace {
-template <typename Metavariables>
-struct component {
-  using metavariables = Metavariables;
-  using chare_type = ActionTesting::MockArrayChare;
-  using array_index = size_t;
-  using simple_tags = tmpl::list<::domain::Tags::FunctionsOfTime>;
-  using phase_dependent_action_list = tmpl::list<
-      Parallel::PhaseActions<
-          typename Metavariables::Phase, Metavariables::Phase::Initialization,
-          tmpl::list<ActionTesting::InitializeDataBox<simple_tags>>>,
-      Parallel::PhaseActions<
-          typename Metavariables::Phase, Metavariables::Phase::Testing,
-          tmpl::list<
-              importers::Actions::ReadSpecThirdOrderPiecewisePolynomial>>>;
-};
-
-struct Metavariables {
-  using component_list = tmpl::list<component<Metavariables>>;
-  enum class Phase { Initialization, Testing, Exit };
-};
-
 void test_options() noexcept {
-  CHECK(db::tag_name<importers::Tags::FunctionOfTimeFile>() ==
-        "FunctionOfTimeFile");
-  CHECK(db::tag_name<importers::Tags::FunctionOfTimeNameMap>() ==
+  CHECK(
+      db::tag_name<domain::FunctionsOfTime::OptionTags::FunctionOfTimeFile>() ==
+      "FunctionOfTimeFile");
+  CHECK(db::tag_name<
+            domain::FunctionsOfTime::OptionTags::FunctionOfTimeNameMap>() ==
         "FunctionOfTimeNameMap");
 
   const std::string option_string{
-      "SpecFuncOfTimeReader:\n"
+      "CubicFunctionOfTimeOverride:\n"
       "  FunctionOfTimeFile: TestFile.h5\n"
       "  FunctionOfTimeNameMap: {Set1: Name1, Set2: Name2}"};
-  using option_tags = tmpl::list<importers::OptionTags::FunctionOfTimeFile,
-                                 importers::OptionTags::FunctionOfTimeNameMap>;
+  using option_tags =
+      tmpl::list<domain::FunctionsOfTime::OptionTags::FunctionOfTimeFile,
+                 domain::FunctionsOfTime::OptionTags::FunctionOfTimeNameMap>;
   Options::Parser<option_tags> options{""};
   options.parse(option_string);
-  CHECK(options.get<importers::OptionTags::FunctionOfTimeFile>() ==
-        "TestFile.h5");
-  const auto& set_names =
-      options.get<importers::OptionTags::FunctionOfTimeNameMap>();
+  CHECK(*(static_cast<std::optional<std::string>>(
+            options.get<
+                domain::FunctionsOfTime::OptionTags::FunctionOfTimeFile>())) ==
+        std::make_optional("TestFile.h5"s));
```

Comment:
```suggestion
CHECK(options.get<domain::FunctionsOfTime::OptionTags::FunctionOfTimeFile>() ==
          "TestFile.h5");
```
(maybe this needs the `std::make_optional`, but I don't think it needs the cast. Similar in the check below.)

Reply:
I only added the static cast because if I don't, I get a compiler error "error: invalid operands to binary expression". I just tried removing the static cast with and without removing the std::make_optional, but neither compiled for me.

If you have a suggested solution that lets me avoid the static cast but still compiles, please let me know, and I'll be happy to make that change

Reply:
Discussed on 11/9 call: looks like the cast is necessary after all

Reply:
Looking closer at this line, I think the conversions are correct, but I don't think you want the `*`.

Reply:
I used the * to check the value of the optional, but  am I right that after moving the *, I'm checking both the value and the bool of the std::optional, instead of just its value?

Reply:
Yep, that's it.